### ISSUE TITLE
Improved support for Pair in dxWDL executor

### ISSUE THREAD

**Issue by michaeltneylon** [EliLillyCo/pytest-wdl#107]
Added some examples here: cc290d8a6e6fb4c20090684219b7e262f6062638

- `dxwdl` executor input formatter is not translating `Array[Pair[X,Y]]` correctly
  - expected: https://github.com/EliLillyCo/pytest-wdl/blob/cc290d8a6e6fb4c20090684219b7e262f6062638/tests/test_dx.py#L192-L199
  - observed: 
  ``` 
  "array_pair_string": [{ 
       "___":  
           { 
               "left": "left_string_val", 
               "right": "right_string_val" 
           } 
       } 
   ] 
  ```
- `dxwdl` executor cannot format workflow_data inputs correctly when it is an `Array[Pair[X, Pair[Y,Z]]]`
  - https://github.com/EliLillyCo/pytest-wdl/blob/cc290d8a6e6fb4c20090684219b7e262f6062638/tests/test_dx.py#L238-L246 errors with
```
self = <pytest_wdl.providers.dx.DxInputsFormatter object at 0x1064bedd8>, path = ('array_pair_pair_string', 'right')

    def _is_wdl_map(self, path: Tuple[str, ...]) -> bool:
        wdl_decls = self._wdl_decls
        path_len = len(path)

        for i, key in enumerate(path, 1):
            is_last = i == path_len

            for name, type_ in wdl_decls.items():
                if name in {None, key}:
                    is_map = isinstance(type_, Type.Map)

                    if is_last:
                        return is_map

                    if isinstance(type_, Type.Array):
                        type_ = cast(Type.Array, type_).item_type

                    if is_map:
                        type_ = cast(Type.Map, type_).item_type[1]
                        # None matches any key
                        wdl_decls: Dict = {None: type_}
                    elif isinstance(type_, Type.StructInstance):
                        wdl_decls = \
                            self._wdl_doc.struct_typedefs[type_.type_name].members

                    break
            else:
                raise ValueError(
>                   f"No input matching key {key} at {path[:i]}"
                )
E               ValueError: No input matching key right at ('array_pair_pair_string', 'right')

pytest_wdl/providers/dx.py:554: ValueError
```
- I also noticed `WDL` module parsing errors in some other existing tests I have that succeed with `pytest-wdl` using the `cromwell` executor and can run with dxWDL/DNAnexus directly. However, they fail to run with `pytest-wdl` `dxwdl` executor currently due to parsing errors around unexpected tokens. I tried to reproduce it but cannot yet, it was around this kind of nested member access:
https://github.com/EliLillyCo/pytest-wdl/blob/cc290d8a6e6fb4c20090684219b7e262f6062638/tests/test_complex.wdl#L34-L38
  - just noting it for now, we can test again later after addressing the above two issues

**Comment by jdidion** [comment]
Thanks. Can you add expected input and output JSON as well?

**Comment by michaeltneylon** [comment]
closed by #108
